### DISCUSSION TITLE
Install Ecole

### DISCUSSION THREAD

**Discussion by zhejiangzhoulei** [ds4dm/ml4co-competition#9] [Q&A] [answered]
Hi,

I followed **Getting started** and installed the ml4co conda environment by running
`source init.sh`

When I tried to evaluate the primal agent of the example by running
`python ../../common/evaluate.py primal item_placement -t 10`

Output
```
  File "../../common/evaluate.py", line 6, in <module>
    import ecole as ec
  File "/opt/anaconda3/envs/ml4co/lib/python3.7/site-packages/ecole/__init__.py", line 3, in <module>
    from ecole.core import RandomEngine, seed, spawn_random_engine, Exception
ImportError: dlopen(/opt/anaconda3/envs/ml4co/lib/python3.7/site-packages/ecole/core.cpython-37m-darwin.so, 2): Symbol not found: __ZN3fmt2v76vprintENS0_17basic_string_viewIcEENS0_17basic_format_argsINS0_20basic_format_contextINS0_8appenderEcEEEE
  Referenced from: /opt/anaconda3/envs/ml4co/lib/python3.7/site-packages/ecole/core.cpython-37m-darwin.so
  Expected in: /opt/anaconda3/envs/ml4co/lib/python3.7/site-packages/ecole/../../../libfmt.8.dylib
 in /opt/anaconda3/envs/ml4co/lib/python3.7/site-packages/ecole/core.cpython-37m-darwin.so
```

It seems that I failed to install ecole. Unfortunately, I did not find the method to address the problem.
Any help or advice would be greatly appreciated. 

Best

**Comment by th-yoon** [comment]
Hi @zhejiangzhoulei , I figured that installing ecole with conda causes this error.
I fixed this error by installing ecole with pip.
For example, in conda.yaml:
```bash
  - pip:
    - ecole==0.7.0.dev1
```

**Reply by zhejiangzhoulei** [reply]
hi @th-yoon , your response is greatly appreciated!

**Comment by gasse** [comment]
Hello @zhejiangzhoulei ,

Could you please provide additional information about your system ? For example:
`uname -a`
`gcc --version`

@th-yoon installing ecole via pip should work as well, although version 0.7.0.dev1 is a bit old and most likely will be incompatible with our evaluation scripts. Either 0.7.0 or 0.7.0.dev3 should work though.

We will investigate on our side as well see if we can spot the issue.

Best,
Maxime

**Comment by gasse** [comment, marked as answer]
@zhejiangzhoulei never mind, I have the error as well. It looks like we messed up the release of v0.7.0. Please use the dev version for now, which you can install as follows:
```bash
conda uninstall ecole -y
conda install -c conda-forge/label/ecole_dev ecole -y
```

We will do a fast release of v0.7.1 which should solve the issue.

Sorry about that !

**Reply by zhejiangzhoulei** [reply]
hi, @gasse. It works! Thank you so much!

**Comment by gasse** [comment]
This issue should be fixed now with `ecole=0.7.1`.

@zhejiangzhoulei @th-yoon  could you please pull the latest version of `submissions/example/init.sh` from the repo, re-install the environment (`sh init.sh`), and try again to make sure the problem is fixed ?

**Reply by zhejiangzhoulei** [reply]
@gasse , yes, the problem is fixed now.

**Reply by gasse** [reply]
Perfect :)

**Reply by th-yoon** [reply]
@gasse Thanks, I was able to install ecole with conda with no problem.
